go/storage/mkvs/db/badger/badger.go

```diff
@@ -891,6 +972,11 @@ func (s *badgerSubtree) PutNode(depth node.Depth, ptr *node.Pointer) error {
 	if err = s.batch.bat.Set(nodeKeyFmt.Encode(&h), data); err != nil {
 		return err
 	}
+	if s.batch.nodes != nil {
+		if err = s.batch.nodes.Set(multipartRestoreNodeLogKeyFmt.Encode(&h), []byte{}); err != nil {
```

Comment:
There is a slight problem with doing this, namely you can only delete nodes in case they did not exist before otherwise you can end up removing existing nodes? You could just check that `ptr.Node.GetCreatedVersion()` is greater than the last version stored in the database (before the restore started) and only add the node in this case?

Hm but there may actually be some other issues when we support restoring checkpoints with pre-existing state in the database (e.g., it won't be actually possible to prune past the gap between the pre-restore version and the restored version). We may need to update `Prune` and add some tests to make sure this works.